### PR TITLE
(CONT-934) Prep for Puppet 8 and Ruby 3

### PR DESCRIPTION
**Pull request by chelnak**
This change updates the default Ruby to 3.2.2 and latest puppet to v8.

The secondary Ruby is now 2.7.8.

### CODE REVIEW

File: configs/components/rubygem-json_pure.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+component "rubygem-json_pure" do |pkg, settings, platform|
+  pkg.version "2.6.3"
+  pkg.sha256sum 'c39185aa41c04a1933b8d66d1294224743262ee6881adc7b5a488ab2ae19c74e'
+
+  instance_eval File.read('configs/components/_base-rubygem.rb')
```
[the remark this answers is not in the document]
Comment (chelnak) [reply configs/components/rubygem-json_pure.rb:5]: I'll give it a go for sure. From memory, we had a couple of issues with JSON which is why pure is here.. 

Hopefully they are gone now!

**Review by joshcooper — APPROVED** [review]
LGTM, though want to try running [vanagon generic builder](https://jenkins-platform.delivery.puppetlabs.net/view/vanagon-generic-builder/job/platform_vanagon-generic-builder_vanagon-generic-builder-init_generic-builder/build?delay=0sec) on all supported pdk `BUILD_TARGETS`?

File: configs/platforms/windows-2019-x64.rb [hunk 2]
```diff
@@ -5,15 +5,16 @@
   visual_studio_version = '2017'
   visual_studio_sdk_version = 'win8.1'
 
+  choco_installer = 'https://chocolatey.org/install.ps1'
+  choco_version = '1.4.0'
   # We need to ensure we install chocolatey prior to adding any nuget repos. Otherwise, everything will fall over
-  plat.add_build_repository "https://artifactory.delivery.puppetlabs.net/artifactory/generic/buildsources/windows/chocolatey/install-chocolatey.ps1"
+  plat.provision_with %(powershell.exe -NoProfile -ExecutionPolicy Bypass -Command '$env:chocolateyVersion="#{choco_version}";iex ((new-object net.webclient).DownloadString("#{choco_installer}"))')
```
Comment (joshcooper) [inline configs/platforms/windows-2019-x64.rb:11]: I ran into this problem too, but we still use win2012R2 with an old .NET version which can't connect to chocolatey.org using TLS 1.2. So I pushed a new `install-chocolatey-1.4.0.ps1` to artifactory and updated our platform definitions to use that https://github.com/puppetlabs/puppet-runtime/pull/679

File: configs/components/rubygem-json_pure.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (david22swan) [reply configs/components/rubygem-json_pure.rb:5]: How did this turn out?

[the remark this answers is not in the document]
Comment (david22swan) [reply configs/components/rubygem-json_pure.rb:5]: Following up on this, it has been agreed that this will be tackled later on as tech debt.

**Review by david22swan — APPROVED** [review]
LGTM

**Pull request merged**